cmd/internal/find/primary/meta.go

```diff
@@ -82,3 +86,440 @@ var metaPrimary = Parser.add(&primary{
+
+  * If {Nu} is specified, then "d" is the difference between v and find's start time.
+    Brackets are useful to distinguish "future" queries from "past" queries.
+
+NOTE: If "d" < 0, then the predicate always returns false. "d" < 0 represents a time
+mismatch (i.e. you are making a past query on a future time value or a future query on a
+past time value). We impose this limitation to make it easier for people to reason about
+time predicates. Otherwise, cases like "-1h" would return true for entries whose meta value
+is a time value in the future despite "-1h" being mentally parsed as "less than one hour
+ago", and thus interpreted as making sense for entries whose time values are in the past.
+
+Below are some examples of time predicates.
+
+  1s
+      Returns true if v was exactly one second ago
+
+  1h
+      Returns true if v was exactly one hour ago
+
+  +1h
+      Returns true if v was more than one hour ago
+
+  -1h
+      Returns true if v was less than one hour ago
+
+  {+1h}
+      Returns true if v is more than one hour from now
+
+  {-1h}
+      Returns true if v is less than one hour from now
+
+And here are some examples of time predicates being used in conjunction with
+object predicates:
+
+  .expiration_date +1h
+      Returns true if m['expiration_date'] was more than one hour ago
+
+  .expiration_date {+1h}
+      Returns true if m['expiration_date'] is more than an hour from now
+
+NOTE: As can be seen from the examples, time predicates only make sense when
+prefixed with a "+" or a "-".
+
+NOTE: As the expiration_date example shows, the "{}" distinguish a "future"
+query from a "past" query. Future queries are a useful way of filtering out
+entries that are going to expire within the next N minutes/hours/days/weeks
+(e.g. security policies, tagged EC2 instances, user credentials, etc.)
+
+STRING PREDICATE:
+Any input that doesn't begin with a "+", "-", "!", "(", or ")" is treated as
+a string predicate. String predicates return false for any non-string values.
+Otherwise, they return true if v == input, where v is the current meta value
+that's being compared.
+
+Below are some examples of string predicates:
+
+  foo
+      Returns true if v == foo 
+
+  bar
+      Returns true if v == bar
+
+And here's an example of a string predicate being used in conjunction with
+an object predicate:
+
+  .owner Mike
+      Returns true if m['owner'] == "Mike"
+
+NOTE: String predicates are a bit limited right now. We plan on adding
+regex/glob support for string predicates, and possibly a way for users to
+specify quoted string literals. The latter's useful when the input begins with
+one of the dis-allowed characters "+", "-", "!", "(", ")".
+
+NEGATION SEMANTICS:
+This section describes each of the aforementioned predicates' negation semantics.
+The information here can be used to reason about input like "! .foo +1".
+
+  OBJECT PREDICATE:  
+  "! .foo p" == ".foo ! p"
```

Comment:
I think object/array predicates could support "-not <predicate_expression>" in addition to "( <predicate_expression> )".